framework/src/transfers/MultiAppTransfer.C

```diff
@@ -274,16 +329,85 @@ MultiAppTransfer::getFromMultiAppInfo()
+{
+  const Real min_x = box.first(0);
+  const Real max_x = box.second(0);
+  const Real min_y = box.first(1);
+  const Real max_y = box.second(1);
+  const Real min_z = box.first(2);
+  const Real max_z = box.second(2);
+
+  std::array<Point, 8> box_corners = {{Point(min_x, min_y, min_z),
+                                       Point(max_x, min_y, min_z),
+                                       Point(min_x, max_y, min_z),
+                                       Point(max_x, max_y, min_z),
+                                       Point(min_x, min_y, max_z),
+                                       Point(max_x, min_y, max_z),
+                                       Point(min_x, max_y, max_z),
+                                       Point(max_x, max_y, max_z)}};
+
+  // transform each corner
+  for (auto & corner : box_corners)
+    corner = transform(corner);
+
+  // Create new bounding box
+  Point new_box_min = box_corners[0];
+  Point new_box_max = new_box_min;
+  for (const auto p : make_range(1, 8))
+    for (const auto d : make_range(Moose::dim))
+    {
+      const Point & pt = box_corners[p];
+      if (new_box_min(d) > pt(d))
+        new_box_min(d) = pt(d);
+
+      if (new_box_max(d) < pt(d))
+        new_box_max(d) = pt(d);
+    }
+  box.first = new_box_min;
+  box.second = new_box_max;
+}
+
+template <typename T>
+void
+expandBoundingBoxes(const Real extension_factor, T & bboxes)
+{
+  if (!extension_factor)
+    return;
+
+  // Expand bounding boxes along all the directions by the same length
+  // Non-zero values of this member may be necessary because the nearest
+  // bounding box does not necessarily give you the closest node/element.
+  // It will depend on the partition and geometry. A node/element will more
+  // likely find its nearest source element/node by extending
+  // bounding boxes. If each of the bounding boxes covers the entire domain,
+  // a node/element will be able to find its nearest source element/node for sure,
+  // but at the same time, more communication will be involved and can be expensive.
+  for (auto & box : bboxes)
   {
-    if (!_from_multi_app->hasLocalApp(i_app))
+    // libmesh set an invalid bounding box using this code
+    // for (unsigned int i=0; i<LIBMESH_DIM; i++)
+    // {
+    //   this->first(i)  =  std::numeric_limits<Real>::max();
+    //   this->second(i) = -std::numeric_limits<Real>::max();
+    // }
+    // If it is an invalid box, we should skip it
+    if (box.first(0) == std::numeric_limits<Real>::max())
       continue;
 
-    _from_local2global_map.push_back(i_app);
-    _from_problems.push_back(&_from_multi_app->appProblemBase(i_app));
-    _from_positions.push_back(_from_multi_app->position(i_app));
+    auto width = box.second - box.first;
+    box.second += width * extension_factor;
+    box.first -= width * extension_factor;
   }
 }
+}
 
 std::vector<BoundingBox>
 MultiAppTransfer::getFromBoundingBoxes()
```

Comment:
this is only called for nearest node, projection and mesh function.
So the bbox_extend_factor parameter is silently ignored for all the others.
Let's suprress it please

Reply:
What do you mean by suppress it?

Reply:
Oh I take your meaning. Since there are only a few classes leveraging those APIs I think that for now I will do the inverse of what you recommend ... I will only add the parameter in the derived classes that use this API

Reply:
works for me